scripts/normalize-by-median.py

```diff
@@ -55,80 +48,94 @@ def WithDiagnostics(ifile, batch_size, fp, paired, norm):
 
     index = 0
 
-    for index, batch in enumerate(batchwise(
-                                  screed.open(ifile, parse_description=False),
-                                  batch_size)):
+    # per read diagnostic output
+    for index, is_paired, read0, read1 in reader:
 
-        norm.total += batch_size
-        total = norm.total
-        discarded = norm.discarded
+        if is_paired:
+            norm.total += 2
+        else:
+            norm.total += 1
 
         if index > 0 and index % 100000 == 0:
             print('... kept {kept} of {total} or {perc:2}%'
-                  .format(kept=total - discarded,
-                          total=total,
-                          perc=int(100. - discarded / float(total) * 100.)),
+                  .format(kept=norm.total - norm.discarded,
+                          total=norm.total,
+                          perc=int(100. - norm.discarded /
+                                   float(norm.total) * 100.)),
                   file=sys.stderr)
 
-            print('... in file ' + input_filename, file=sys.stderr)
+            print('... in file ' + ifilename, file=sys.stderr)
 
-            if report_fp:
+            if fp:
                 print(total + " " + total - discarded + " " +
                       1. - (discarded / float(total)), file=fp)
                 report_fp.flush()
 
-        # If in paired mode, check that the reads are properly interleaved
-        if paired:
-            if not check_is_pair(batch[0], batch[1]):
-                raise IOError('Error: Improperly interleaved pairs \
-                    {b0} {b1}'.format(b0=batch[0].name, b1=batch[1].name))
+        yield index, is_paired, read0, read1
+
+    # per file diagnostic output
+    if norm.total == 0:
+        print('SKIPPED empty file ' + ifilename, file=sys.stderr)
+    else:
+        print('DONE with {inp}; kept {kept} of {total} or {perc:2}%'
+              .format(inp=ifilename, kept=norm.total - norm.discarded,
+                      total=norm.total, perc=int(100. - norm.discarded /
+                                                 float(norm.total) * 100.)),
+              file=sys.stderr)
 
-        yield batch
+    if fp:
```

Comment:
Please de-indent following block.

Also, please use some form of string formatting, either str.format or '%' here -- str.format is fine.  But don't do str(a) + ... + str(b).